### ISSUE TITLE
04_https_request certificate format incorrect

### ISSUE THREAD

**Issue by ggsubs** [espressif/esp-idf#112]
The server cert file included in the example is in an incorrect format. The file is pem, the example expects der encoded. The cert file can be converted by:

```openssl x509 -inform pem -in server_root_cert.pem -outform der -out server_root_cert.der  ```

Once converted component.mk should be updated as well as the extern ref to the object:

```
extern const uint8_t server_root_cert_pem_start[] asm("_binary_server_root_cert_der_start");
extern const uint8_t server_root_cert_pem_end[]   asm("_binary_server_root_cert_der_end");
```

**Comment by negativekelvin** [comment]
Example works for me as is.

**Comment by projectgus** [comment]
What caused you to think that the example code expects a DER formatted certificate?

The certificate data is passed to the mbedTLS function `mbedtls_x509_crt_parse`, which supports both PEM and DER format certificates provided that support is enabled in the mbedTLS configuration file. We should have support for both in the default `esp_config.h` configuration in esp-idf.

DER format certificates are a computationally simpler to parse (no decoding step), but PEM is a lot easier to work with which is why I chose it for the example.

**Comment by ggsubs** [comment]
When the pem format used mbedTls returns a format error:
```
I (396) example: Seeding the random number generator
I (406) example: Loading the CA root certificate...
E (406) example: mbedtls_x509_crt_parse returned -0x2180
```
The example only worked when replaced the cert with der format.

**Comment by projectgus** [comment]
Thanks for clarifying. I don't get this failure on the current esp-idf master 8ab4e11:

```
I (207) example: Seeding the random number generator
I (207) example: Loading the CA root certificate...
I (217) example: Setting hostname for TLS session...
I (217) example: Setting up the SSL/TLS structure...
```

Maybe a clean build might help, or there's some other modification or config tweak which will allow me to reproduce?

Angus

**Comment by igrr** [comment]
We have reproduced this on OS X. @ggsubs which OS are you running?

**Comment by igrr** [comment]
Fixed in 5ac7810.
